src/vs/workbench/api/electron-browser/mainThreadSaveParticipant.ts

```diff
@@ -37,6 +37,26 @@ import { ICodeActionsOnSaveOptions } from 'vs/editor/common/config/editorOptions
 import { IBulkEditService } from 'vs/editor/browser/services/bulkEditService';
 import { CancellationTokenSource } from 'vs/base/common/cancellation';
 
+// {{SQL CARBON EDIT}}
+import { INotebookService } from 'sql/workbench/services/notebook/common/notebookService';
+
+class NotebookUpdateParticipant implements ISaveParticipantParticipant {
```

Comment:
Please comment on the purpose of this participant. E.g.
```
/*
 * An update participant that ensures any untracked changes are synced to the JSON file contents for a 
 * Notebook before save occurs. While every effort is made to ensure model changes are notified and a listener 
 * updates the backing model in-place, this is a backup mechanism to hard-update the file before save in case 
 * some are missed.
 */
```